File: web/src/api/userConfig.ts
// ── 用户 AI 配置（per-user）· API 层 ──────────────────────────────────
// /api/me/ai-config：读取（key 掩码）/ 更新 / 测试连接。
import { api } from './client'

export interface AiConfig {
  provider: string
  model: string
  base_url: string
  api_key_masked: string   // sk-****1234；空表示未设置
  has_key: boolean
  max_turns: number
  max_budget_usd: number
  system_prompt_extra: string | null
  enabled: boolean
  configured: boolean
}

export interface AiConfigUpdate {
  provider?: string
  model?: string
  base_url?: string
  api_key?: string         // 留空 / 含 **** → 后端保留旧值
  max_turns?: number
  max_budget_usd?: number
  system_prompt_extra?: string | null
  enabled?: boolean
}

export interface AiConfigTestResult {
  ok: boolean
  message: string
  model: string | null
  latency_ms: number | null
}

export function getAiConfig(): Promise<AiConfig> {
  return api<AiConfig>('/me/ai-config')
}

export function updateAiConfig(data: AiConfigUpdate): Promise<AiConfig> {
  return api<AiConfig>('/me/ai-config', { method: 'PUT', body: data })
}

/** 测试连接：可不传（测已保存配置）或传未保存的表单值。 */
export function testAiConfig(data?: AiConfigUpdate): Promise<AiConfigTestResult> {
  return api<AiConfigTestResult>('/me/ai-config/test', { method: 'POST', body: data ?? {} })
}

File: server/app/schemas/user_config.py
from pydantic import BaseModel, Field


class AiConfigResponse(BaseModel):
    provider: str = "anthropic"
    model: str = ""
    base_url: str = ""
    api_key_masked: str = ""        # sk-****1234；空表示未设置
    has_key: bool = False
    max_turns: int = 12
    max_budget_usd: float = 0.5
    system_prompt_extra: str | None = None
    enabled: bool = True
    configured: bool = False        # model/base_url/key 是否齐备


class AiConfigUpdate(BaseModel):
    provider: str | None = None
    model: str | None = Field(default=None, max_length=64)
    base_url: str | None = Field(default=None, max_length=255)
    # 明文 key；None / 空 / 含 **** 的掩码串 → 保留旧值（不覆盖）
    api_key: str | None = Field(default=None, max_length=512)
    max_turns: int | None = Field(default=None, ge=1, le=100)
    max_budget_usd: float | None = Field(default=None, ge=0, le=100)
    system_prompt_extra: str | None = Field(default=None, max_length=4000)
    enabled: bool | None = None


class AiConfigTestResult(BaseModel):
    ok: bool
    message: str
    model: str | None = None
    latency_ms: int | None = None
